codesign/lib/src/transfer_utils.dart

```diff
@@ -0,0 +1,92 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:file/file.dart';
+import 'package:process/process.dart';
+
+import './utils.dart';
+
+/// Utility function class to handle upload/download of files from/to google cloud.
+class EngineArtifactTransfer {
+  EngineArtifactTransfer({
+    required this.gsCloudBaseUrl,
+    this.uploadFunction,
+    this.downloadFunction,
+  }) {
+    uploadFunction ??= defaultUploadFunction;
```

Comment:
This is a bad practice. It's hard to debug and be sure what's used in prod vs tests.

Prefer injecting the services needed instead of the functions being used. In this case, you only need to pass `ProcessManager` and `FileSystem`